src/uucore/src/lib/features/checksum.rs

```diff
@@ -513,7 +512,16 @@ where
                     .unwrap_or((String::new(), None))
                 } else if let Some(a) = algo_name_input {
                     // When a specific algorithm name is input, use it and use the provided bits
-                    (a.to_lowercase(), length_input)
+                    // except when dealing with blake2b, where we will detect the length
+                    if algo_name_input == Some(ALGORITHM_OPTIONS_BLAKE2B) {
+                        let length = match expected_checksum.len() {
+                            128 | 96 | 64 | 48 | 32 => Some(expected_checksum.len() / 2),
```

Comment:
I don't understand what you are doing on this line. Why do you divide by 2? And why are you only interested in some selected lengths?

Reply:
selected length was a mistake :)
i added a comment about /2